packages/core/src/components/advanced-select/advanced-select.tsx

```diff
@@ -0,0 +1,160 @@
+import { Component, ComponentInterface, Prop, h, FunctionalComponent, Event, EventEmitter } from "@stencil/core";
+
+import {
+  AdvancedSelectGroupRedirect,
+  AdvancedSelectOption,
+  AdvancedSelectOptionsOrGroup,
+} from "./advanced-select.models";
+import {
+  AdvancedSelectClickEvent,
+  AdvancedSelectOptionClickEvent,
+  AdvancedSelectRedirectClickEvent,
+} from "./advanced-select.interfaces";
+
+@Component({
+  tag: "dso-advanced-select",
+  styleUrl: "advanced-select.scss",
+  shadow: true,
+})
+export class AdvancedSelect implements ComponentInterface {
+  /**
+   * The options to display in the select.
+   */
+  @Prop()
+  options: AdvancedSelectOptionsOrGroup<any>[] = [];
+
+  /**
+   * The active option. By object reference.
+   */
+  @Prop()
+  active?: AdvancedSelectOption<any>;
+
+  /**
+   * An extra text for the active option. Only visible in the list of options.
+   */
+  @Prop()
+  activeHint?: string;
+
+  /**
+   * The open state of the options list.
+   */
+  @Prop()
+  open: boolean = false;
+
+  /**
+   * Emitted when user clicks the select.
+   */
+  @Event({ bubbles: false })
+  dsoClick!: EventEmitter<AdvancedSelectClickEvent>;
+
+  /**
+   * Emitted when user clicks an option
+   */
+  @Event({ bubbles: false })
+  dsoOptionClick!: EventEmitter<AdvancedSelectOptionClickEvent<any>>;
+
+  /**
+   * Emitted when user clicks a redirect link.
+   */
+  @Event({ bubbles: false })
+  dsoRedirectClick!: EventEmitter<AdvancedSelectRedirectClickEvent>;
+
+  private handleClick = (event: MouseEvent) => {
+    this.dsoClick.emit({ originalEvent: event });
+  };
+
+  private handleOptionClick = (event: MouseEvent, option: AdvancedSelectOption<any>) => {
+    this.dsoOptionClick.emit({ originalEvent: event, option });
+  };
+
+  private handleRedirectClick = (event: MouseEvent, redirect: AdvancedSelectGroupRedirect) => {
+    this.dsoRedirectClick.emit({ originalEvent: event, redirect });
+  };
+
+  render() {
+    return (
+      <div>
+        <button class={{ "active-option": true, open: this.open }} type="button" onClick={this.handleClick}>
```

Comment:
Stencil's DOM renderer heeft karige ondersteuning voor `class` vergeleken met Angular's `ngClass`. Dit gemis hebben we opgevangen met `clsx`:

```suggestion
        <button class={clsx(["active-option", { open: this.open }])} type="button" onClick={this.handleClick}>
```

Reply:
Wist niet dat dat de reden was voor het gebruik van `clsx`, weet ik dat ook vanaf nu. Ik ga het aanpassen 👍🏻

Reply:
🤙